tap_google_search_console/streams/abstract.py

```diff
@@ -0,0 +1,357 @@
+                            max_bookmark_value = utils.strftime(bookmark_dt_tm)
+
+                        # Keep only records whose bookmark is after the last_datetime
+                        if bookmark_dt_tm >= last_dt_tm:
+                            write_record(self.tap_stream_id, transformed_record, time_extracted=time_extracted)
+                            counter.increment()
+                    else:
+                        write_record(self.tap_stream_id, transformed_record, time_extracted=time_extracted)
+                        counter.increment()
+
+            LOGGER.info(f"Stream: {self.tap_stream_id}, Processed {counter.value} records")
+            self.records_extracted += counter.value
+            return max_bookmark_value
+
+    def get_records_for_sub_type(self, site_url: str, sub_type: str, state: Dict, schema: Dict, stream_metadata: Dict) -> None:
+        """
+        Sync the data for a given sub_type, stream, site
+        Gets the bookmark value or start date value, extracts data for date window size of 30 days
+        """
+        start_dt_tm, end_dt_tm = self.set_start_and_end_times(state, site_url, self.tap_stream_id, sub_type)
+        LOGGER.info(f"bookmark value or start date for {self.tap_stream_id} {site_url} {sub_type}: {start_dt_tm}")
+        site_path = encode_and_format_url(site_url, self.path)
+        while start_dt_tm < end_dt_tm:
+            offset, row_limit, batch_count = 0, self.row_limit, self.row_limit
+            last_datetime = self.get_bookmark(state, self.tap_stream_id, site_url, sub_type,
+                                              self.config.get("start_date"))
+            bookmark_value = last_datetime
+            start_str, end_str = utils.strftime(start_dt_tm)[:10], utils.strftime(end_dt_tm)[:10]
+
+            LOGGER.info(f"Running sync for {site_url}, {self.tap_stream_id}, {sub_type} between date window "
+                        f"{start_str} {end_str}")
+            payload = self.make_payload(sub_type, start_str, end_str, stream_metadata)
+            while row_limit == batch_count:
+                body = {"startRow": offset, "rowLimit": row_limit, **payload}
+                time_extracted = utils.now()
+                LOGGER.info(f"body = {body}")
+                data = self.client.post(site_path, endpoint=self.tap_stream_id, data=json.dumps(body))
+                if not data:
+                    self.write_bookmark(state, site_url, sub_type, bookmark_value)
+                    start_dt_tm, end_dt_tm = self.modify_start_end_dt_tm(end_dt_tm)
+                transformed_data = []
+                if self.data_key in data:
+                    transformed_data = transform_json(data, self.tap_stream_id, self.data_key, site_url, sub_type,
+                                                      dimensions_list=payload.get('dimensions', []))[self.data_key]
+                else:
+                    LOGGER.info("Number of raw data records: 0")
```

Comment:
Is this else block really required? You can write the logger in if block itself.

Reply:
Moved this to If condition